feat(website): auto generate oxfmt docs

Boshen opened the pull request:
Corresponding PR: https://github.com/oxc-project/oxc-project.github.io/pull/643

The next PR will get the json schema auto generated as well.

leaysgur reviewed (changes requested):
```sh
❯ cargo run -p website_formatter cli

error[E0609]: no field `external_formatter` on type `&FormatService`
  --> apps/oxfmt/src/service.rs:98:21
   |
98 |             || self.external_formatter.is_none()
   |                     ^^^^^^^^^^^^^^^^^^ unknown field
   |
   = note: available fields are: `allocator_pool`, `cwd`, `output_options`, `format_options`

error[E0308]: `if` and `else` have incompatible types
   --> apps/oxfmt/src/service.rs:101:16
    |
 96 |           let formatted = if cfg!(not(feature = "napi"))
    |  _________________________-
 97 | |             || self.format_options.embedded_language_formatting.is_off()
 98 | |             || self.external_formatter.is_none()
    | |________________________________________________- `if` and `else` have incompatible types
 99 |           {
100 |               base_formatter.format(&ret.program)
    |               ----------------------------------- expected because of this
101 |           } else {
    |  ________________^
102 | |             #[cfg(feature = "napi")]
103 | |             {
104 | |                 let Some(external_formatter) = &self.external_formatter else {
...   |
110 | |         };
    | |_________^ expected `Formatted<'_>`, found `()`

Some errors have detailed explanations: E0308, E0609.
For more information about an error, try `rustc --explain E0308`.
error: could not compile `oxfmt` (lib) due to 2 previous errors
```

As a result, nothing is printed. I will fix this first in another PR.

leaysgur commented:
(Git operations failed, currently performing recovery operations...)

leaysgur reviewed (approved):
Will merge after CI.

Also, https://github.com/oxc-project/oxc-project.github.io/pull/643

leaysgur commented:
### Merge activity

* **Nov 26, 3:45 AM UTC**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.com/merges?org=oxc-project&repo=oxc) once it meets the requirements.
* **Nov 26, 3:45 AM UTC**: `leaysgur` added this pull request to the [Graphite merge queue](https://app.graphite.com/merges?org=oxc-project&repo=oxc).

Boshen committed:
feat(website): auto generate oxfmt docs (#15985)

Corresponding PR: https://github.com/oxc-project/oxc-project.github.io/pull/643

The next PR will get the json schema auto generated as well.
